test/react-native/features/steps/react-native-steps.rb

```diff
@@ -55,3 +55,12 @@
   valid_values = table.raw.flatten
   assert_true(valid_values.include?(actual_value), "#{field_path} value: #{actual_value} did not match the given list: #{valid_values}")
 end
+
+Then("the payload fields match one of the following sets:") do |data_table|
```

Comment:
I think moving to this step loses us some assurances compared with the individual assertions before:

- sometimes the order is important
- we now no longer are guaranteed that every entry in the table was used – the same input could have been received each time and would still pass

I think if we were to use something like this, it would be better to collect all the requests and say "given assertions xyz, do I have a request that matches each?"